aries_cloudagent/connections/models/diddoc_v2/tests/test_diddoc.py

```diff
@@ -0,0 +1,195 @@
+        assert did_instance.service[0].serialize() == service
+        assert did_instance.service[0] == service_instance
+
+    async def test_add_new_verification_method(self):
+        did = {"id": "did:sov:LjgpST2rjsoxYegQDRm7EL", "publicKey": [publicKey]}
+        publicKey_instance = PublicKey.deserialize(publicKey)
+        did_instance = DIDDoc(id=did["id"], publicKey=[publicKey_instance])
+        assert did_instance.id == did["id"]
+        assert len(did_instance.publicKey) == 1
+        assert did_instance.publicKey[0].serialize() == publicKey
+        assert did_instance.publicKey[0] == publicKey_instance
+
+        publicKey2 = publicKey
+        publicKey2["id"] = "did:sov:LjgpST2rjsoxYegQDRm7EL#5"
+        publicKey_instance2 = PublicKey.deserialize(publicKey2)
+        did_instance.set(publicKey_instance2)
+        assert len(did_instance.publicKey) == 2
+
+    async def test_add_new_verification_method(self):
+        did = {"id": "did:sov:LjgpST2rjsoxYegQDRm7EL", "publicKey": [publicKey]}
+        publicKey_instance = PublicKey.deserialize(publicKey)
+        did_instance = DIDDoc(id=did["id"], publicKey=[publicKey_instance])
+        assert did_instance.id == did["id"]
+        assert len(did_instance.publicKey) == 1
+        assert did_instance.publicKey[0].serialize() == publicKey
+        assert did_instance.publicKey[0] == publicKey_instance
+
+        publicKey2 = publicKey
+        publicKey2["id"] = "did:sov:LjgpST2rjsoxYegQDRm7EL#5"
+        publicKey_instance2 = PublicKey.deserialize(publicKey2)
+        did_instance.set(publicKey_instance2)
+        assert len(did_instance.publicKey) == 2
+
+    async def test_add_new_verification_method(self):
+        did = {"id": "did:sov:LjgpST2rjsoxYegQDRm7EL", "publicKey": [publicKey]}
+        publicKey_instance = PublicKey.deserialize(publicKey)
+        did_instance = DIDDoc(id=did["id"], publicKey=[publicKey_instance])
+        assert did_instance.id == did["id"]
+        assert len(did_instance.publicKey) == 1
+        assert did_instance.publicKey[0].serialize() == publicKey
+        assert did_instance.publicKey[0] == publicKey_instance
+
+        publicKey2 = publicKey
+        publicKey2["id"] = "did:sov:LjgpST2rjsoxYegQDRm7EL#5"
+        publicKey_instance2 = PublicKey.deserialize(publicKey2)
+        did_instance.set(publicKey_instance2)
+        assert len(did_instance.publicKey) == 2
+
+    async def test_deserialize_ok(self):
+        did = {
+            "id": "did:sov:LjgpST2rjsoxYegQDRm7EL",
+            "service": [service],
+            "publicKey": [publicKey],
+            "authentication": [publicKey],
+        }
+
+        result = DIDDoc.deserialize(did)
+
+        service_instance = result.dereference("did:sov:LjgpST2rjsoxYegQDRm7EL#2")
+        publicKey_instance = result.dereference("did:sov:LjgpST2rjsoxYegQDRm7EL#3")
+        assert service_instance.serialize() == service
+        assert publicKey_instance.serialize() == publicKey
```

Comment:
It would be good to see a little more thorough testing of the `dereference` method separate from just testing deserialization. Explicit assertions about the type of the object returned be dereferencing would be good too, i.e. `assert isinstance(service_instance, Service)`.